contracts/ApplicationRegistry.sol

```diff
@@ -148,8 +125,8 @@ contract ApplicationRegistry is Ownable, Pausable {
             ApplicationState.Submitted
         );
         applicantGrant[msg.sender][_grant] = true;
-        grantRef.incrementApplicant();
         emit ApplicationSubmitted(_id, _grant, msg.sender, _metadataHash, _milestoneCount, block.timestamp);
+        grantRef.incrementApplicant();
```

Comment:
why are we emitting events before the state change? I see this at couple other places too.

Reply:
Actually it was raised by the tool, however events will only be emitted if the transaction is successfull, in case the transaction reverts events wont be emitted